### PR TITLE
Rebuild RabbitMQ node and trigger

### PR DESCRIPTION
**Pull request by Ken-Michalak**
* Use amqp-connection-manager to automatically reconnect
* Support options on messages
* Add ability to bind a queue to an exchange
* Use prefetch on queues to prevent overloading n8n

### CODE REVIEW

**Comment by Ken-Michalak** [comment]
This should fix #1988 and #2806.

**Comment by mirobertod** [comment]
Any update on this? We also need the feature to bind a queue to an exchange.
Thanks

**Comment by mirobertod** [comment]
@janober can you take a look at this? Thanks!

File: packages/nodes-base/tsconfig.json [hunk 1]
```diff
@@ -11,6 +11,7 @@
 		"module": "commonjs",
 		"removeComments": true,
 		"strict": true,
+		"skipLibCheck": true,
```
Comment (michael-radency) [inline packages/nodes-base/tsconfig.json:14]: why did you add this option? it has some caveats and does not work well with monorepo projects

**Comment by michael-radency** [comment]
Hello @Ken-Michalak thank you for your contribution, I did open new PR https://github.com/n8n-io/n8n/pull/3244 as I was denied of permission when pushing some fixes

File: packages/nodes-base/tsconfig.json (hunk 1 above)
Comment (Ken-Michalak) [reply packages/nodes-base/tsconfig.json:14]: There were some type-related errors coming from dependencies, and it tends to be a pretty standard thing to use.
https://github.com/tsconfig/bases/blob/main/bases/node14.json

I'm not sure what other options there are besides making a pr for the packages that have issues.

Comment (michael-radency) [reply packages/nodes-base/tsconfig.json:14]: Hi @Ken-Michalak I did work today on your PR without this option and did not got any errors, can you please check changes that I made in https://github.com/n8n-io/n8n/pull/3244
